pkg/engine/utils/exceptions.go

```diff
@@ -96,7 +112,7 @@ func checkResourceDescription(
 	subresource string,
 ) bool {
 	if len(conditionBlock.Kinds) > 0 {
-		if !matched.CheckKind(conditionBlock.Kinds, gvk, subresource, true) {
+		if !matchesResource(conditionBlock.Kinds, gvk.Kind, subresource) {
```

Comment:
We need to match full GVK and subresource instead of kind only. Is there an issue in `matched.CheckKind` for matching subresources?